### DISCUSSION TITLE
Opening Mac Mouse Fix & Malware Messages

### DISCUSSION THREAD

**Discussion by noah-nuebling** [noah-nuebling/mac-mouse-fix#114] [Guides]
If you encountered a problem with opening the Mac Mouse Fix app, this guide will help you solve it.

> [!NOTE]
> **Read this guide if you're on Mac Mouse Fix 2.2.3 or below**
> Mac Mouse Fix 2.2.4 and later are 'notarised' by Apple, which should prevent the issues described in this guide. \
> If you experience problems with opening Mac Mouse Fix 2.2.4 or later, please leave a comment below, so I can help you out and update this guide. Thank you. :)

### 1. Problem: 'Cannot check for malicious software' message

When trying to open Mac Mouse Fix for the first time, you might encounter this message:
<img width="350" align="center" src="https://user-images.githubusercontent.com/15073177/117338109-7a02c600-ae9e-11eb-91c5-2dee38ae1d7c.png">

**Why does this message appear?**

In macOS, there is a security feature called 'Notarization'. When an app is notarized, that means Apple has checked it and determined it to be safe. 
Unfortunately, it costs $99 to notarize an app, making it a suboptimal option for small open-source projects like Mac Mouse Fix. 
So Mac Mouse Fix is not notarized, that's why this message appears.

However, since the Mac Mouse Fix source code is freely available and thousands of people are using it and have the ability to write about their experiences on this GitHub page, you can still be sure that Mac Mouse Fix is safe.

**Update:** Mac Mouse Fix 2.2.4 and later are notarized by Apple, so this message should not appear anymore. \
I wrote a little bit about the rationale behind notarizing the app in the  [2.2.4](https://github.com/noah-nuebling/mac-mouse-fix/releases/tag/2.2.4) release notes.

**The solution**

To open Mac Mouse Fix anyways:
1. Right-click Mac Mouse Fix in the Finder, then click 'Open'
2. Click 'Open' again in the window that appears

Mac Mouse Fix should then open normally and the 'cannot check for malicious software' message won't appear anymore.

---

### 2. Problem: 'Will damage your computer' message

If you are already a Mac Mouse Fix user, you might have encountered messages saying that Mac Mouse Fix is malware when trying to open Mac Mouse Fix or after restarting your computer:

<img width="600" align="center" src="https://user-images.githubusercontent.com/232541/117108938-7b58c380-adb6-11eb-9497-b4503161249b.png">

**Why do these messages appear?**

Mac Mouse Fix version 1.0.3 and below were signed with a 'code signing certificate' which has since been revoked. That's what prompted the messages.
1.0.4 and above were signed with a new certificate, but after the 2.0 release, that certificate was also revoked. 

[2.1](https://github.com/noah-nuebling/mac-mouse-fix/releases/tag/2.1.0) and later are signed with a new, valid certificate. 

See #95 to learn more about the issue.

**The solution**

Simply [download](http://noah-nuebling.github.io/mac-mouse-fix-website) the latest version of Mac Mouse Fix. It should open normally.

---

I hope this helped you! If you have any questions you can:
- Write a comment down below
- [Open a new GitHub Discussion](https://github.com/noah-nuebling/mac-mouse-fix/discussions)
- [Send me an Email](mailto:noah.n.public@gmail.com?)
  - Please note that I get many emails and I don't have that much time, so I might take a long time to respond

General feedback and improvement ideas for this guide are also very welcome of course!

**Comment by tiagofmartinho** [comment]
Right-clicking the icon and clicking open does not work. It simply shows that the software needs to be updated and the only options are to show the app in Finder or press OK. :(

**Reply by sjdennis** [reply]
Are you clicking the icon in the dock? You need to right-click the icon in the applications folder (ie in Finder), not in the dock.

If you right-click the icon in the dock, it will helpfully give you the option of "show in Finder", to take you straight to the app in the applications folder, so you can then right-click it there and open it as described above. A little convoluted and confusing, but it's just how it works. Apple seems to be trying to push people away from any software they don't earn money from, by making this confusing.

**Comment by amadeuspzs** [comment]
For anyone that's struggling try the following referenced in #95 

`xattr -d com.apple.quarantine /Applications/Mac\ Mouse\ Fix.app`

**Reply by tomsotte** [reply]
@noah-nuebling can you put this answer in the first post as well? At least for macOS 12 this is the only solution that worked

**Reply by dmares01** [reply]
This fixes the issue on Ventura 13.6 as well

**Comment by whoave** [comment]
I am facing this issue:
![image](https://user-images.githubusercontent.com/39208904/148650166-e0e24f51-9884-479a-94ae-a9f64ef2d0c2.png)
In English:
Mac Mouse Fix unexpectedly interrupted.
Click "Report" to get more detailed information and notify Apple.

Ignore, Report...

Mac version: 10.15.7

**Reply by whoave** [reply]
![image](https://user-images.githubusercontent.com/39208904/148650324-4e05d0f6-d139-4a48-967c-ad5a5a71525f.png)

**Reply by noah-nuebling** [reply]
Hi @whoave, thanks for reaching out! Those logs were very helpful in figuring out what's going on. [Version 2.1](https://github.com/noah-nuebling/mac-mouse-fix/releases) should resolve your issue and it even brings a neat new feature!

**Comment by yunuserkaya** [comment]
the app is completely broken now

**Reply by yunuserkaya** [reply]
<img width="372" alt="image" src="https://user-images.githubusercontent.com/22533048/148657627-a86fa268-5a81-46bf-9ba0-1a0bc2f1d328.png">
<img width="453" alt="image" src="https://user-images.githubusercontent.com/22533048/148657633-c51deafd-8e6a-4aa4-b516-aba1e5c3595a.png">

**Comment by franciscosucre** [comment]
I cannot use the app anymore. It just says it will damage my computer. Using 2.0 :(

**Comment by noah-nuebling** [comment]
Hey guys, the certificate has just been revoked again causing the malware messages :(

I'm on it!

**Reply by franciscosucre** [reply]
Thanks bro, i am way to used to your app now. Thanks for all your hard work

**Reply by noah-nuebling** [reply]
[Version 2.1](https://github.com/noah-nuebling/mac-mouse-fix/releases) is out now! It should resolve this issue.

**Reply by franciscosucre** [reply]
thanks man, you rock! I can be productive with my mouse again!

**Comment by lzm0** [comment]
Tried every approach, including disabling the Gatekeeper.  Ended up with `The application "Mac Mouse Fix" can't be opened` even for an old version. I just built it from source code on my own.

**Reply by yunuserkaya** [reply]
The one I built crashes on launch. Do you mind sharing the one you built here?

`Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: @rpath/ZipArchive.framework/Versions/A/ZipArchive
Referenced from: /Applications/Mac Mouse Fix.app/Contents/MacOS/Mac Mouse Fix
Reason: tried: '/Applications/Mac Mouse Fix.app/Contents/MacOS/../Frameworks/ZipArchive.framework/Versions/A/ZipArchive' (code signature in <0434EF6D-6F8A-3FA2-9A1A-7C9B86683C62> '/Applications/Mac Mouse Fix.app/Contents/Frameworks/ZipArchive.framework/Versions/A/ZipArchive' not valid for use in process: mapped file has no Team ID and is not a platform binary (signed with custom identity or adhoc?)), '/Applications/Mac Mouse Fix.app/Contents/MacOS/../Frameworks/ZipArchive.framework/Versions/A/ZipArchive' (code signature in <0434EF6D-6F8A-3FA2-9A1A-7C9B86683C62> '/Applications/Mac Mouse Fix.app/Contents/Frameworks/ZipArchive.framework/Versions/A/ZipArchive' not valid for use in process: mapped file has no Team ID and is not a platform binary (signed with custom identity or adhoc?)), '/Library/Frameworks/ZipArchive.framework/Versions/A/ZipArchive' (no such file
(terminated at launch; ignore backtrace)`

**Reply by noah-nuebling** [reply]
I think saw similar errors when I disabled signing.
 
Make sure your "Signing" settings look like this except for the "Team" on all targets:

<img width="500px" src="https://user-images.githubusercontent.com/40808343/148662165-1f3418cc-26b3-40c8-9dcb-ce2590ff3236.png">

Hope that helps!

**Reply by yunuserkaya** [reply]
it is working Noah, thanks so much!

**Reply by noah-nuebling** [reply]
Awesome! No worries at all. By the way, I just published [version 2.1](https://github.com/noah-nuebling/mac-mouse-fix/releases). It should work normally again and it has a cool new feature.

**Reply by yunuserkaya** [reply]
whoa whoa whoa now I can change screen brightness with mouse 4 and 5? Thank you!

**Reply by noah-nuebling** [reply]
Yepp :)

**Comment by yunuserkaya** [comment]
I'm not sure I like middle-click+drag anymore. Perhaps we need a disable? I'm having a hard time closing tabs quickly because the mouse registers a drag after middle clicking a tab instance.

**Reply by noah-nuebling** [reply]
Hey there Yunus, thanks for the feedback! You can disable the 'Click and Drag' Action by selecting it and then clicking the '-' button in the bottom left of the Action Table.

In the latest version 2.2.0, I also made it a little harder to accidentally use the Click and Drag feature. 

Hope that helps! Let me know if you have any more questions.

**Reply by noah-nuebling** [reply]
Oh and you could also assign Click and Drag to a side button! I find that more comfy anyways.

**Comment by jdub850000** [comment]
Hi. Great app, but even with the new 2.1, I'm getting Apple unidentified developer and possible malware errors? I've also had a couple app crashes while trying to assign button 4 on my mouse. Also the middle click and drag setting for middle click and drag (Blender) keeps reverting to Mission Control & Spaces - I've got a crash report if you're interested?

**Reply by noah-nuebling** [reply]
Hey @jdub850000 thanks for reaching out, I'm glad you like the app!

The 'Cannot check for malicious software' message is expected even under the newest version. To avoid it I'd have to pay money to Apple for a developer account. Is that the message you saw? (Screenshot above)

If it crashed when you chose the secret "Middle Click and Drag" option, that's actually expected as well. I started building that feature but noticed I'd have to restructure some stuff pretty deep in the app to make it useful, and so I gave up on it for now. I just forgot to remove the option. I want to provide better compatibility with Blender in the future though, and already have plans for that. For now you'll unfortunately just have to [uncapture](https://github.com/noah-nuebling/mac-mouse-fix/discussions/112) the Middle Mouse Button to use it with Blender or disable Mac Mouse Fix while you're using Blender. 

Hope that helps!

If it crashes in another situation, then it'd be great if you could send me the steps to reproduce or the crash reports you got. Thanks!

**Comment by airjohnpro** [comment]
This is happening again since updating to macOS 12.3 today.

**Reply by airjohnpro** [reply]
My company's IT team just reached out to let me know it flagged their anti-virus software. It made me delete the app last night. I let them know I was in contact with the developer. Good news is, they are all excited to try the app!

**Reply by noah-nuebling** [reply]
Hey John, thanks for reaching out! :)

I have a lot of stuff to do for college, so I don't have too much time to tend to MMF at the moment. But if you could tell me what Antivirus software is flagging MMF, I'll look into that and see what I can do! Thanks!

**Reply by airjohnpro** [reply]
Jamf Protect, thank you!

**Reply by noah-nuebling** [reply]
Hmm there's not much I can do about this I'm afraid, but I contacted Jamf about it. Hopefully they can address it soon!

**Reply by airjohnpro** [reply]
@noah-nuebling My IT team came back and said it is actually Gatekeeper (12.2/12.2.1) flagging it:

> Gatekeeper prevented an application from running. A user attempted to run a program that macOS Gatekeeper is specifically programed to prevent.

**Reply by airjohnpro** [reply]
They share the json file from the console as well below. Thanks!

  "host": {
    "ips": [
      "192.168.1.9"
    ],
    "serial": "FVFF61CWQ6LW",
    "hostname": "FVFF61CWQ6LW",
    "provisioningUDID": "00008103-0001793E229A001E"
  },
  "match": {
    "tags": [
      "Gatekeeper"
    ],
    "uuid": "45EC14AF-C8F6-41C0-9A3B-597BF66AF145",
    "event": {
      "pid": 3234,
      "name": "Mac Mouse Fix",
      "path": "/Applications/Mac Mouse Fix.app/Contents/MacOS/Mac Mouse Fix",
      "uuid": "91506958-174D-4382-BD8A-7BAE4594B219",
      "sender": "AppleSystemPolicy",
      "process": "kernel",
      "category": "",
      "subsystem": "",
      "timestamp": 1643252511.063571,
      "composedMessage": "ASP: Security policy would not allow process: 3234, /Applications/Mac Mouse Fix.app/Contents/MacOS/Mac Mouse Fix",
      "senderImagePath": "/System/Library/Extensions/AppleSystemPolicy.kext/Contents/MacOS/AppleSystemPolicy",
      "processImagePath": "/kernel",
      "processIdentifier": 0
    },
    "facts": [
      {
        "name": "GatekeeperBlockedRevoked",
        "tags": [
          "Gatekeeper"
        ],
        "uuid": "B83111E8-9F75-419A-8513-D625C6B64CB4",
        "human": "Gatekeeper blocked a revoked app",
        "actions": [
          {
            "name": "Report",
            "parameters": {}
          }
        ],
        "context": [],
        "version": 1,
        "severity": 2
      }
    ],
    "actions": [],
    "context": [],
    "severity": 2
  },
  "related": {
    "files": [],
    "users": [
      {
        "uid": 503,
        "name": "john.wiskowski",
        "uuid": "FVFF61CWQ6LW1f7"
      }
    ],
    "groups": [
      {
        "gid": 20,
        "name": "staff",
        "uuid": "FVFF61CWQ6LW14"
      }
    ],
    "binaries": [
      {
        "gid": 20,
        "uid": 503,
        "fsid": 16777234,
        "mode": 33261,
        "path": "/Applications/Mac Mouse Fix.app/Contents/MacOS/Mac Mouse Fix",
        "size": 523744,
        "inode": 4476563,
        "xattrs": [
          "com.apple.quarantine"
        ],
        "changed": 1620417886,
        "created": 1620273092,
        "sha1hex": "3162859e53167946d81c8cc8864235fe61a87db8",
        "accessed": 1643250998,
        "modified": 1620273092,
        "sha256hex": "4087940d9ef52604aa959adb0c398fc5c4e13357f7437a9177c37b7c05762810",
        "isDownload": false,
        "isAppBundle": false,
        "isDirectory": false,
        "signingInfo": {
          "appid": "com.nuebling.mac-mouse-fix",
          "cdhash": "AAAAAAAAAAAAAAAAAAAAAAAAAAA=",
          "status": -2147409652,
          "teamid": "LM5Z78756B",
          "signerType": 3,
          "authorities": [],
          "entitlements": [],
          "statusMessage": "CSSMERR_TP_CERT_REVOKED"
        },
        "isScreenShot": false
      }
    ],
    "processes": []
  },
  "eventType": "GPGatekeeperEvent"
}

**Reply by airjohnpro** [reply]
My IT team followed up with JAMF and this response worked for me. Sharing as an FYI and happy I can again use your amazing app!
`I did test with this app and received a gatekeeper warning, where I have to right click and open it for the first time to bypass the warning, but I didn't get any alerts on my test device running Protect and it didn't block me from using it with the default built in threat prevention as well as no custom threat prevention lists set up.`

**Reply by noah-nuebling** [reply]
Awesome. Thanks so much for the kind words!

**Comment by whoave** [comment]
There is a problem which if I add Click and Drag for Middle mouse (for example), it disabled systems default main middle click event. I cannot click with my middle button, nothing happens. If I add middle click to mac mouse fix, and assign it to middle click action, it works as a default middle click button but it works with a delay, I mean as default when I click with middle button it works fast, but with mac mouse fix's middle mouse click event, it takes time to do what middle button does. For example if I click a link with middle click after assigning it to middle click on the app, it takes couple seconds to open the link in another tab.

**Reply by noah-nuebling** [reply]
Hi Mert! If you remove the 'Double Click' action, the delay should disappear. Hope that helps, let me know if you have more questions!

**Reply by whoave** [reply]
Actually I wasn't using double click action, I was using click, hold and click&drag actions but after sending the message above, I wonder if there is an update and installed 2.2.0 and its fixed I think. I don't have delay with these 3 active. I think you already fixed the issue when you updated the threshold for click drag event. Thanks again ✋🏻

**Reply by noah-nuebling** [reply]
Okay cool, I'm glad it works for you now!!

**Reply by whoave** [reply]
No :D I was wrong yes its about double click, I think I was using double click. Not hold. So yeah, I removed double click and will use hold instead for mission control :D

**Comment by ahmetsali** [comment]
Hi Noah, thanks for this incredibly well-working fix. I have just switched from windows to mac and your app fixed everything about the mouse. I also liked the smooth scrolling but it scrolls too slowly, even when the scrolling speed is maxed out. It would be nice to see faster scrolling.

Btw, I am using another app called "mos" for smooth scrolling, which has a nice faster smooth scrolling feature.

**https://mos.caldis.me/**

Keep up the good work!

**Reply by whoave** [reply]
@noah-nuebling I agree with the post above, I am using Mos for smooth scrolling too. Mac mouse fix's button customization is best but scrolling is not good enough. @noah-nuebling if you download and see how Mos works and implement the same functionality, there won't be any need for an other application than Mac mouse fix for all our mouse needs.

**Comment by banx9x** [comment]
Hi, after updating os to MacOS 13.0 Public Beta, I cannot enable Mac Mouse Fix (click on checkbox does nothing)

**Comment by Unfaced85** [comment]
Hi there, new to the Mac World here and wanted to use this app as I read lots of good things about it. unfortunately, on MAcOS12.4 and 12.5 I keep getting this 'Cannot check for malicious software' message even though I followed the Control Click open method - still happens. Any suggestions would be great, thank you.

PS: as I am kind of a Mac n00b, please make any suggestions followable for newbies, thanks ;)

**Reply by tomsotte** [reply]
Try this, it should work on macOS 12 https://github.com/noah-nuebling/mac-mouse-fix/discussions/114#discussioncomment-1123763

**Reply by Unfaced85** [reply]
thanks, did not work...... until I realized I need to adapt to this: 
xattr -d com.apple.quarantine ./Downloads/Mac\ Mouse\ Fix.app

so thanks

**Comment by pay4UU** [comment]
Hallo, I have very archaic iMac with OSX 10,11

when i tried to start MMF from GUI - it did not start and without response or GUI warning/error message . Starting from bash it did following:

/Applications/Mac Mouse Fix.app/Contents$ MacOS/Mac\ Mouse\ Fix 
2022-10-03 13:07:14.035 Mac Mouse Fix[4233:103221] Invalid color System, separatorColor (warning given only once)
Segmentation fault: 11

**Comment by agabek-ov** [comment]
Hi everyone!

After reading all the comments, I couldn't fix the problem on my M1 Pro MacBook, even though I've used this app on my previous MacBook with no problem while opening from Finder.

So, here's the link to fix the problem from the terminal:
https://www.androidauthority.com/macos-cannot-verify-that-this-app-is-free-from-malware-3192659/

Might be helpful to someone:)

**Comment by Eddie-Hartman** [comment]
Also had this issue pop up on M2 Air on Ventura 13.1.

**Comment by medo30886** [comment]
Hi All, just got my mouse Rapoo MT750s and all is great with mouse fix thanks a lot however the secondary scroll is not readable by the mouse fix hence i can not use it, anyway that i can get it to be editable / readable

**Comment by idyl1026** [comment]
It's perfect, especially if you can grab the gesture buttons on the MX Master 3

**Comment by jlourencGit** [comment]
@noah-nuebling security is important. And I personally believe users shouldn’t be taught to ignore those warnings. BUT I also believe Apple should give it for free for open source. To make a long story short, contact me via DM and I’ll be more than happy to help with notarization

**Comment by mgeeeee7** [comment]
Hey Noah, the app is great. I've got a Logitech mouse with a button that's not being recognized. Any workarounds?

Thanks

**Reply by warmand00** [reply]
I just struggled with the same problem today! For me I'm using a Logitech G600 and the Middle Click (mouse scroll wheel press) was not being recognized. When I turn on Logitech G Hub software, it works perfectly. I now have the G Hub software to open on startup. As soon as G Hub was turned on, all of the middle-click functions enabled/enhanced by Mac Mouse Fix worked flawlessly.

**Reply by mgeeeee7** [reply]
Interesting. I ended up installing Mac Mouse Fix to not have to deal with Logitech Options + software, which was using 4gigs of ram (to run a mouse?) as well as tons of CPU and energy. It was wildly terrible.

**Reply by warmand00** [reply]
Wow that's crazy! Mine is taking up 113MB RAM. I have no idea what could make G Hub run wild like that
